### DISCUSSION TITLE
WebView2 says: "Extensions are not enabled"

### DISCUSSION THREAD

**Discussion by peakpeak-github** [MicrosoftEdge/WebView2Feedback#4793] [Q&A]
WebView2 replies with "Extensions are not enabled" when calling 
profile7->AddBrowserExtension(extensionPath.c_str(), Callback( ....);

How can I enable extensions in WebView2?

**Comment by jls28** [comment]
options6->put_AreBrowserExtensionsEnabled(TRUE)

**Comment by peakpeak-github** [comment]
Thanks.  
Before:
options6->get_AreBrowserExtensionsEnabled(pb); says FALSE
After running options6->put_AreBrowserExtensionsEnabled(TRUE);
options6->get_AreBrowserExtensionsEnabled(pb); says TRUE

I still get the same result and message: 'Extensions are not enabled'

**Reply by jls28** [reply]
Create CoreWebView2EnvironmentOptions,
call options->put_AreBrowserExtensionsEnabled(TRUE)
just before CreateCoreWebView2EnvironmentWithOptions :
this work for me.

**Comment by peakpeak-github** [comment]
I had done just about that but changed the code to your example. Same result. I'm not using profiles in the rest of my code but I'm using tabs. Can that affect the outcome?
Adding an extension works in the sample WebView2APISample where there are profiles but no tabs.
It would speed up the troubleshooting if one could step through the core code of AddBrowserExtension() and see what it is complaining about.
